plugins/firehydrant/src/components/ServiceDetailsCard/ServiceDetailsCard.tsx

```diff
@@ -0,0 +1,279 @@
+    textAlign: 'center',
+    justifyItems: 'center',
+    width: '102px',
+    backgroundColor: theme.palette.type === 'dark' ? '#f1edff' : '',
+    '&:hover, &:focus': {
+      backgroundColor: '#f1edff',
+      color: '#614ab6',
+    },
+    '&:active': {
+      color: '#3b2492',
+      backgroundColor: '#b2a6e3',
+      boxShadow:
+        'rgb(59, 36, 146) 0px 0px 0px 1px inset, rgb(141, 134, 188) 3px 3px 0px 0px inset;',
+    },
+    border: '1px solid #3b2492',
+    borderRadius: '5px',
+    padding: '10px',
+  },
+  buttonLink: {
+    backgroundColor: '#3b2492',
+    color: '#FFF',
+    textTransform: 'none',
+    '&:hover': {
+      backgroundColor: '#614ab6',
+    },
+  },
+  buttonContainer: {
+    display: 'grid',
+    gridGap: '24px',
+    gridAutoFlow: 'column',
+    gridAutoColumns: 'min-content',
+  },
+  icon: {
+    color: '#f1642d',
+  },
+  link: {
+    textDecoration: 'underline',
+    fontSize: '16px',
+    lineHeight: '27px',
+    color: '#3b2492',
+    '&:hover, &:focus': {
+      fontWeight: '500',
+    },
+  },
+  linksContainer: {
+    borderBottom: '1px solid #d5d5d5',
+    padding: '10px 0px 10px 20px',
+    backgroundColor: '#f1edff',
+    marginBottom: '20px',
+  },
+  table: {
+    width: '100%',
+  },
+  warning: {
+    display: 'flex',
+    alignItems: 'center',
+    padding: '10px',
+    background: '#f1edff',
+    color: theme.palette.type === 'dark' ? '#3b2492' : '#fff',
+  },
+}));
+
+const ServiceWarning = ({ text }: { text: string }) => {
+  const classes = useStyles();
+  return (
+    <div className={classes.warning}>
+      <WarningIcon />
+      &nbsp;&nbsp;<span>{text}</span>
+    </div>
+  );
+};
+
+const ServiceAnalyticsView = ({
+  serviceId,
+  startDate,
+  endDate,
+}: {
+  serviceId: string;
+  startDate: moment.Moment;
+  endDate: moment.Moment;
+}) => {
+  const {
+    loading: analyticsLoading,
+    value: analyticsValue = {},
+    error: analyticsError,
+  } = useServiceAnalytics({
+    serviceId,
+    startDate: startDate.format('YYYY-MM-DD'),
+    endDate: endDate.format('YYYY-MM-DD'),
+  });
+
+  return (
+    <ServiceAnalytics
+      loading={analyticsLoading}
+      value={analyticsValue}
+      error={analyticsError}
+    />
+  );
+};
+
+export const ServiceDetailsCard = () => {
+  const { entity } = useEntity();
+  const classes = useStyles();
+  const [showServiceDetails, setShowServiceDetails] = useState(false);
+  const configApi = useApi(configApiRef);
+
+  const BASE_URL =
+    configApi.getOptionalString('firehydrant.baseUrl') ||
+    'https://app.firehydrant.io';
+
+  const startDate = moment().subtract(30, 'days').utc();
+  const endDate = moment().utc();
+
+  const { loading, value, error } = useServiceDetails({
+    serviceName: entity?.metadata?.name,
+  });
+
+  const activeIncidents: string[] = value?.service?.active_incidents ?? [];
+  const incidents: ServiceIncidentsResponse = value?.incidents ?? [];
+  const serviceId: string = value?.service?.id!;
```

Comment:
I'm wondering if this should be an annotation on the entity rather than using the name of the entity to look up in the FireHydrant API. Names are not unique across namespaces, so you might not get the results you desire.

You could use `useEntity` hook like above, but look for `annotations['firehydrant.io/service-id']` or something instead and use that to search through FH?

Reply:
That's a good call out and the non-unique names was something we considered (FireHydrant API also supports non-unique service names). I think I understand what your ask is here, to make it less brittle using an ID via annotations, however FireHydrant API doesn't have a unique backstage ID to lookup in our database in the first place. The only way to look up the record is through the service-name. Any more thoughts here?

Reply:
Yep the edit sounds about right. Feel free to choose the name of the annotation such that it makes sense / sounds familiar to an FH user. You mention the term "service name" several times; if that's a familiar wording and it's an ID like string and users tend to think in terms of it - it's probably a good annotation!